add go-autocomplete and company-go packages

Funnnny opened the pull request:
I removed go-autocomplete extension and add package from melpa instead (I don't think it's there at my last PR but today I found it, don't know why)

Also I added support for company-go.

syl20bnr commented on contrib/lang/go/packages.el:
```diff
@@ -31,3 +33,19 @@ which require an initialization must be listed explicitly in the list.")
     (add-hook 'go-mode-hook 'go-eldoc-setup))
+
+(defun go/init-go-autocomplete()
+  (use-package go-autocomplete
```
Is it possible to lazy load it with `:defer t` or with a hook in `:init` ?

syl20bnr commented on contrib/lang/go/packages.el:
```diff
@@ -31,3 +33,19 @@ which require an initialization must be listed explicitly in the list.")
+(defun go/init-company-go ()
+ (use-package company-go
+   :if (boundp 'company-backends)
+   :defer t
```
This package may never load because there is no hook and only a `:config` property which is called after the package has been loaded.

Funnnny commented:
I found some problem with go-mode as well, it seems to load even in python mode.

I will read a bit about use-package and update the PR later

syl20bnr commented:
Great, don't hesitate to ask questions here.

I plan to write a crash course for lazy loading stuff relevant for spacemacs in the documentation. I hope to be able to do it soon.

Funnnny commented:
So I changed the set key so it won't set in other mode, also I moved the ac and company backend to init (like in lang/python contrib).

Please let me know if I need to do anything.

syl20bnr commented:
Great ! :+1: 
Cherry-picked in `develop` branch, you can safely delete your branch.

Thank you !

Note: prefer rebasing instead of merging the last changes of develop in your PR branch. To replay your PR commits on top of develop branch do:
```sh
$ git checkout go-company
$ git rebase develop
```
